src/wmrc/validate.py

```diff
@@ -0,0 +1,167 @@
+    """
+
+    previous_year = current_year - 1
+    if current_year not in metrics_df.index:
+        raise ValueError(f"Current year {current_year} not found in index")
+    if previous_year not in metrics_df.index:
+        raise ValueError(f"Previous year {previous_year} not found in index")
+
+    diffs = metrics_df.loc[current_year] - metrics_df.loc[previous_year]
+    pct_change = diffs / metrics_df.loc[previous_year] * 100
+
+    #: have to rename columns after pct_change so that the indices line up properly for the calculation
+    diffs.columns = [f"{col}_diff" for col in diffs.columns]
+    pct_change.columns = [f"{col}_pct_change" for col in pct_change.columns]
+
+    values_current_year = metrics_df.loc[current_year].copy().reindex(index=diffs.index)
+    values_current_year.columns = [f"{col}_{current_year}" for col in values_current_year.columns]
+
+    values_previous_year = metrics_df.loc[previous_year].copy().reindex(index=diffs.index)
+    values_previous_year.columns = [f"{col}_{previous_year}" for col in values_previous_year.columns]
+
+    everything = pd.concat([pct_change, values_current_year, values_previous_year, diffs], axis=1)
+
+    return everything[
+        list(interleave([pct_change.columns, values_current_year.columns, values_previous_year.columns, diffs.columns]))
+    ]
+
+
+def run_validations():
+
+    #: Get records from salesforce, run summary methods
+    wmrc_skid = Skid()
+    records = wmrc_skid._load_salesforce_data()
+    _ = records.deduplicate_records_on_facility_id()
+    facility_summary_df = summarize.facility_metrics(records)
+    county_summary_df = summarize.counties(records)
+
+    #: Calc year-over-year changes
+    facility_changes = facility_year_over_year(facility_summary_df, records.df, 2023)
+    county_changes = county_year_over_year(county_summary_df, 2023)
+    state_changes = state_year_over_year(county_summary_df, 2023)
+
+    #: Remove county-wide and statewide prefixes so we can concat the different change dfs by row
+    county_changes.rename(
+        columns={col: col.replace("county_wide_", "") for col in county_changes.columns}, inplace=True
+    )
+    state_changes.rename(columns={col: col.replace("statewide_", "") for col in state_changes.columns}, inplace=True)
+
+    all_changes = pd.concat([facility_changes, county_changes, state_changes], axis=0)
+
+    #: Move the msw_recycling_rate columns to the front, write to csv
+    index_a = all_changes.columns.get_loc("msw_recycling_rate_pct_change")
+    slice_b = all_changes.columns.slice_indexer("msw_recycling_rate_pct_change", "msw_recycling_rate_diff")
+    index_c = all_changes.columns.get_loc("msw_recycling_rate_diff") + 1
+    new_index = all_changes.columns[slice_b].append([all_changes.columns[:index_a], all_changes.columns[index_c:]])
+
+    all_changes.reindex(columns=new_index).to_csv(r"c:\gis\projects\wmrc\data\from_sf\validation_2.csv")
```

Comment:
**question**: could this be made relative to the file to remove the need for your systems file structure?

Reply:
Given the infrequent running of this script, I find it easier to open it in VSCode, change the vars, and run it from there instead of giving it a proper cli that would allow us to pass the output path in. Maybe I'm just being lazy, I dunno.

Reply:
Actually, thinking about trying to remember when to run it, I wonder if it wouldn't be worth creating a second function and schedule to trigger it. That way the first report would be automatic, and if they need other reports it's as simple as logging into the console and kicking off the task again. Thoughts @steveoh? would a second function cost too much?

Reply:
Yeah, as more time passes we forget more and more and these intricacies become less familiar. I like to help future me and make these things as simple as possible. It seems to make up for the investment now.